.changesets/implement-heartbeats.md

```diff
@@ -0,0 +1,44 @@
+---
+bump: "patch"
+type: "add"
+---
+
+Add heartbeats support. You can use the `heartbeat` function to send heartbeats
+directly from your code, to track the execution of certain processes:
+
+```python
+from appsignal import heartbeat
+
+def send_invoices():
+  # ... your code here ...
+  heartbeat("send_invoices")
+```
+
+You can use the `Heartbeat` class (not the `heartbeat` function!) with a
+`with` block, to report to AppSignal both when the process starts, and when
+it finishes, allowing you to see the duration of the process:
+
+```python
+from appsignal import Heartbeat
+
+def send_invoices():
+  with Heartbeat("send_invoices"):
```

Comment:
That I can use both the capitalized and non-capitalized way is going to trip up in the future for sure. Is this a common pattern?

Reply:
Thank you for pointing this out! I see the problem. It is unclear to me how to meaningfully address it. Some thoughts:

You can also use both the capitalized class and non-capitalized methods in the other integrations as well -- alongside the `heartbeat` convenience method, the `Appsignal.Heartbeat` class/module in Ruby/Elixir/Node.js is exposed for people who want to call start and finish manually, in case that's an use case that exists. 

But an important difference is that this is the only integration in which we _document_ that, because, unlike in the other integrations, one of the convenience APIs (the context manager one) is tied to the class.

I'd say it's not super uncommon to have a convenience API with simple function calls and a more flexible API based around classes. The `requests` library comes to mind: usually you just do `requests.get(...)`, but you can also create a `requests.Request(object)`. It is probably more uncommon to have them use the same name.

---

Anyway. The way to address it that comes to mind is to not document the context manager API. Document Python's function-based API, like it is done for Elixir and Node.js, and omit documenting the Ruby-style block-based API.

This saves us a lot of "that's `heartbeat` not `Heartbeat`" in the docs. We can always re-evaluate if customers reach out saying they'd like it to be exposed as a context manager.

Does this seem reasonable for now?

Reply:
Yeah I'd be okay with simplying it a bit. Or to make it more distinct, like `Heartbeat.new()` vs `heartbeat()`

Reply:
`Heartbeat.new()` is unfortunately a Ruby-ism -- class instance initialisation in Python is always `Heartbeat()`

Amended the changelog to remove references to the uppercase-h `Heartbeat`.